metpy/plots/skewt.py

```diff
@@ -275,12 +278,20 @@ def __init__(self, fig=None, rotation=30, subplot=(1, 1, 1)):
             fig = plt.figure(figsize=figsize)
         self._fig = fig
 
-        # Handle being passed a tuple for the subplot, or a GridSpec instance
-        try:
-            len(subplot)
-        except TypeError:
-            subplot = (subplot,)
-        self.ax = fig.add_subplot(*subplot, projection='skewx', rotation=rotation)
+        if rect is None and subplot is None:
```

Comment:
Not wild about the complexity in argument handling introduced here...but have no good ways to improve it either.

Reply:
Yeah, I'm not sure how to resolve that, either. One suggestion might be something like this:

```python
    def __init__(self, fig=None, rotation=30, position=(1, 1, 1)):
        # ...
        if len(position) == 3:
            # Assume position represents a subplot
        elif len(position) == 4:
            # Assume position represents a rectangle
        else:
            # Raise an error
```
That seems more complex and ripe for confusion than the current implementation, though. Plus the interface would not be backward compatible.

I added the exception raise to prevent a situation in which the user passes both arguments and wonders why it's not respecting the rect argument, but maybe that's not necessary.

Reply:
Maybe a way to clean it up a bit, only one double conditional:

```python
if rect and subplot:
    raise ValueError("Specify only one of `rect' and `subplot', but not both")

elif rect:
     self.ax = fig.add_axes(rect, projection='skewx', rotation=rotation)

elif subplot:
    # Handle being passed a tuple for the subplot, or a GridSpec instance
    try:
        len(subplot)
    except TypeError:
        subplot = (subplot,)
     self.ax = fig.add_subplot(*subplot, projection='skewx', rotation=rotation)

else:  # Nothing specified, assume a simple single panel plot
     subplot = (1, 1, 1)
```

Reply:
@jrleeman In your suggestion, the `else` clause doesn't do anything, but in general I like your approach. How about:
```python
if rect and subplot:
    raise ValueError("Specify only one of `rect' and `subplot', but not both")

elif rect:
     self.ax = fig.add_axes(rect, projection='skewx', rotation=rotation)

elif subplot:
    # Handle being passed a tuple for the subplot, or a GridSpec instance
    try:
        len(subplot)
    except TypeError:
        subplot = (subplot,)
else:  # Nothing specified, assume a simple single panel plot
     subplot = (1, 1, 1)

self.ax = fig.add_subplot(*subplot, projection='skewx', rotation=rotation)
```

Reply:
@dopplershift, the last line would generate an error if called with a rect. Perhaps this?

```python
if rect and subplot:
    raise ValueError("Specify only one of `rect' and `subplot', but not both")

elif rect:
    self.ax = fig.add_axes(rect, projection='skewx', rotation=rotation)

else:
    # Handle being passed a tuple for the subplot, or a GridSpec instance
    if subplot:
        try:
            len(subplot)
        except TypeError:
            subplot = (subplot,)
    else:
         subplot = (1, 1, 1)
    self.ax = fig.add_subplot(*subplot, projection='skewx', rotation=rotation)
```

Reply:
Ugh, you're right. Your version looks good--assuming it runs without error. 😈 This is why we have tests. 😁

Reply:
Looks like it does indeed finish the tests with no [additional] errors.